scipio/src/io/dma_file.rs

```diff
@@ -412,7 +412,7 @@ impl DmaFile {
         let new_path = new_path.as_ref().to_owned();
         let old_path = path_required!(self, "rename")?;
         enhanced_try!(
-            crate::io::rename_file(old_path, &new_path).await,
```

Comment:
Hm, there’s an extra to_owned above still.

Reply:
This one is correct in my view.
Note that right below this line we're storing new_path as a copy.

We could not do `.to_owned()` in the beginning and do it closer to the place where we store a copy, but that becomes a style issue.

Reply:
well, this is really minor but if we move the `to_owned` closer to its user we save an allocation in the failure case.
so let's do that.